obspy/io/rg16/util.py

```diff
@@ -50,9 +50,11 @@ def _read(fi, position, length, dtype, left_part=True):
         return _read_bcd(fi, length, left_part)
     elif dtype == 'binary':
         return _read_binary(fi, length, left_part)
-    elif dtype == 'IEEE':
-        data = np.frombuffer(fi.read(int(length)), '>f4')
-        return data[0] if len(data) == 1 else data
+    if dtype == 'IEEE':
+        dtype = '>f4'
+    # dtype should be understood by numpy, dispatch to frombuffer
+    data = np.frombuffer(fi.read(int(length)), dtype)
```

Comment:
shouldn't this return be in the if previously, and raise an exception if dtype is not known?

Reply:
Thanks for pointing this out, the docs for this function are a little out of date. The function is actually correct because it allows ints and other numpy dtypes to also be used. I updated the docs string to be more clear.